File: 283. Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        length = len(nums)
        while i < length:
            if nums[i] == 0:
                length -= 1
                nums.pop(i)
                nums.append(0)
            else:
                i += 1

File: 283. Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0;
        int length = nums.size();
        while (i < length)
        {
            if (nums[i] == 0)
            {
                nums.erase(nums.begin() + i);
                nums.push_back(0);
                length --;
            }
            else
                i ++;
        }
    }
};
